pkg/ast/ast_node.go

```diff
@@ -115,23 +115,44 @@ func (d *Document) NodeDirectives(node Node) []int {
 	}
 }
 
-func (d *Document) RemoveDirectiveFromNode(node Node, ref int) {
+func (d *Document) RemoveDirectivesFromNode(node Node, directiveRefs []int) {
+	for _, ref := range directiveRefs {
+		d.RemoveDirectiveFromNode(node, ref)
+	}
+}
+
+func (d *Document) RemoveDirectiveFromNode(node Node, directiveRef int) {
 	switch node.Kind {
 	case NodeKindFragmentSpread:
-		if i, ok := d.IndexOf(d.FragmentSpreads[node.Ref].Directives.Refs, ref); ok {
+		if i, ok := d.IndexOf(d.FragmentSpreads[node.Ref].Directives.Refs, directiveRef); ok {
 			d.FragmentSpreads[node.Ref].Directives.Refs = append(d.FragmentSpreads[node.Ref].Directives.Refs[:i], d.FragmentSpreads[node.Ref].Directives.Refs[i+1:]...)
```

Comment:
as this is a slice trick and we are using it a lot maybe we could move it to a `helpers.go`
smth like
```go
// delete - is a delete slice trick with preserving order of items
func delete(arr *[]int, index int) {
  arr = append(arr[:index], arr[index+1:]...)
}
```

and we will be able to simplify a code a bit
```go
d.FragmentSpreads[node.Ref].Directives.Refs = delete(&d.FragmentSpreads[node.Ref].Directives.Refs, i)
```